File: VB/09_Charts/SetChartDataNumberFormat.vb
Imports System.ComponentModel
Imports System.Text
Imports Spire.Presentation
Imports Spire.Presentation.Charts

Namespace SetChartDataNumberFormat
	Partial Public Class Form1
		Inherits Form
		Public Sub New()
			InitializeComponent()
		End Sub

		Private Sub btnRun_Click(ByVal sender As Object, ByVal e As EventArgs) Handles btnRun.Click
			'Create PPT document and load file
			Dim ppt As New Presentation()
			ppt.LoadFromFile("..\..\..\..\..\..\Data\SetChartDataNumberFormat.pptx")

			'Get chart on the first slide
			Dim chart As IChart = TryCast(ppt.Slides(0).Shapes(0), IChart)

			'Set the number format for Axis
			chart.PrimaryValueAxis.NumberFormat = "#,##0.00"

			'Set the DataLabels format for Axis
			chart.Series(0).DataLabels.LabelValueVisible = True
			chart.Series(0).DataLabels.PercentValueVisible = False
			chart.Series(0).DataLabels.NumberFormat = "#,##0.00"
			chart.Series(0).DataLabels.HasDataSource = False

			'Set the number format for ChartData
			For i As Integer = 1 To chart.Series(0).Values.Count
				chart.ChartData(i, 1).NumberFormat = "#,##0.00"
			Next i

			Dim result As String = "SetChartDataNumberFormat_output.pptx"

			'Save the document
			ppt.SaveToFile(result, FileFormat.Pptx2013)

			'Launch the result file
			PPTDocViewer(result)
		End Sub

		Private Sub PPTDocViewer(ByVal fileName As String)
			Try
				Process.Start(fileName)
			Catch
			End Try
		End Sub
	End Class
End Namespace

File: CS/09_Charts/SetChartDataNumberFormat.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Text;
using System.Windows.Forms;
using Spire.Presentation;
using Spire.Presentation.Charts;

namespace SetChartDataNumberFormat
{
    public partial class Form1 : Form
    {
        public Form1()
        {
            InitializeComponent();            
        }

        private void btnRun_Click(object sender, EventArgs e)
        {
            //Create PPT document and load file
            Presentation ppt = new Presentation();
            ppt.LoadFromFile(@"..\..\..\..\..\..\Data\SetChartDataNumberFormat.pptx");

            //Get chart on the first slide
            IChart chart = ppt.Slides[0].Shapes[0] as IChart;

            //Set the number format for Axis
            chart.PrimaryValueAxis.NumberFormat = "#,##0.00";

            //Set the DataLabels format for Axis
            chart.Series[0].DataLabels.LabelValueVisible = true;
            chart.Series[0].DataLabels.PercentValueVisible = false;
            chart.Series[0].DataLabels.NumberFormat = "#,##0.00";
            chart.Series[0].DataLabels.HasDataSource = false;

            //Set the number format for ChartData
            for (int i = 1; i <= chart.Series[0].Values.Count; i++)
            {
                chart.ChartData[i, 1].NumberFormat = "#,##0.00";
            }

            String result = "SetChartDataNumberFormat_output.pptx";

            //Save the document
            ppt.SaveToFile(result, FileFormat.Pptx2013);

            //Launch the result file
            PPTDocViewer(result);
        }

        private void PPTDocViewer(string fileName)
        {
            try
            {
                System.Diagnostics.Process.Start(fileName);
            }
            catch { }        
        }
    }
}